addons/mail/models/mail_mail.py

```diff
@@ -83,6 +83,24 @@ def default_get(self, fields):
     scheduled_date = fields.Char('Scheduled Send Date',
         help="If set, the queue manager will send the email after the date. If not set, the email will be send as soon as possible. Unless a timezone is specified, it is considered as being in UTC timezone.")
 
+    @api.model
+    def fields_get(self, *args, **kwargs):
+        # related selection will fetch translations from DB
+        # selections added in stable won't be in DB -> add them on the related model if not already added
+        message_type_field = self.env['mail.message']._fields['message_type']
+        if ('auto_comment', 'Automated Targeted Notification') not in message_type_field.get_description(self.env)['selection']:
+            self._fields_get_message_type_update_selection(message_type_field.selection)
+        return super().fields_get(*args, **kwargs)
+
+    def _fields_get_message_type_update_selection(self, selection):
+        """Update the field selection for message type on mail.message to match the runtime values.
+
+        DO NOT USE it is only there for a stable fix and should not be used for any reason other than hotfixing.
+        """
+        self.env['ir.model.fields'].invalidate_cache(['selection_ids'])
+        self.env['ir.model.fields.selection']._update_selection('mail.message', 'message_type', selection)
```

Comment:
No problem with regular users, they have access to those models ? mail.mail is limited but hey, asking for a friend